plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/kubernetes/KubernetesLabelConverter.java

```diff
@@ -0,0 +1,130 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.openshift.client.kubernetes;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.che.plugin.docker.client.json.ContainerConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Converter of labels defined in {@link ContainerConfig} for matching to Kubernetes
+ * annotation requirements
+ */
+public class KubernetesLabelConverter {
+    private static final Logger LOG = LoggerFactory.getLogger(KubernetesLabelConverter.class);
+    /** Prefix used for che server labels */
+    private static final String CHE_SERVER_LABEL_PREFIX  = "che:server";
+    /** Padding to use when converting server label to DNS name */
+    private static final String CHE_SERVER_LABEL_PADDING = "0%s0";
+    /** Regex to use when matching converted labels -- should match {@link CHE_SERVER_LABEL_PADDING} */
+    private static final Pattern CHE_SERVER_LABEL_KEY    = Pattern.compile("^0(.*)0$");
+    private static final String KUBERNETES_ANNOTATION_REGEX = "([A-Za-z0-9][-A-Za-z0-9_\\.]*)?[A-Za-z0-9]";
+
+    /**
+     * @return prefix that is used for Che server labels
+     */
+    public String getCheServerLabelPrefix() {
+        return CHE_SERVER_LABEL_PREFIX;
+    }
+
+    /**
+     * Converts a map of labels to match Kubernetes annotation requirements. Annotations are limited
+     * to alphanumeric characters, {@code '.'}, {@code '_'} and {@code '-'}, and must start and end
+     * with an alphanumeric character, i.e. they must match the regex
+     * {@code ([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]}
+     *
+     * <p>Note that entry keys should begin with {@link KubernetesLabelConverter#CHE_SERVER_LABEL_PREFIX} and
+     * entries should not contain {@code '.'} or {@code '_'} before conversion;
+     * otherwise label will not be converted and included in output.
+     *
+     * <p>This implementation is relatively fragile -- changes to how Che generates labels may cause
+     * this method to stop working. In general, it will only be possible to convert labels that are
+     * alphanumeric plus up to 3 special characters (by converting the special characters to {@code '_'},
+     * {@code '-'}, and {@code '.'} as necessary).
+     *
+     * @param labels Map of labels to convert
+     * @return Map of labels converted to DNS Names
+     */
+    public Map<String, String> labelsToNames(Map<String, String> labels) {
+        Map<String, String> names = new HashMap<>();
+        for (Map.Entry<String, String> entry : labels.entrySet()) {
```

Comment:
SonarLint is flagging this loop as error
```
Loops should not contain more than a single "break" or "continue" statement
squid:S135 
Restricting the number of break and continue statements in a loop is done in the interest of good structured programming.
One break and continue statement is acceptable in a loop, since it facilitates optimal coding. If there is more than one, the code should be refactored to increase readability.
```